### PR TITLE
Re-architect ABC.

### PR DESCRIPTION
**Pull request by robyurkowski**
This is good to go, I think. There are still a few issues but they're not to do with the rearching:
- The Menu Presenter tries to call `to_html` on its menu's children, when it should call it on presenters wrapping its menu's children;
- `MockMenuEntry` classes inside the Presenter classes are conflicting. Temporarily solved by renaming one to `MenuEntryMock`, but someone with more testing experience could give me some guidance here;
- Doc needs to be updated (which I'll do shortly).
- Travis should probably be updated, though since it's not working at the moment, I'm feeling rather laconic about getting that done.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Inspired heavily by Spree's build.sh. Thanks, Spree! You rock.
```
Comment (knewter) [inline build.sh:1]: :heart:

File: delivery/rails/spec/presenters/html/menu_entry_presenter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          subject.menu_entry.children.count.should == 1
+        end
+
+        it "renders its children" do
+          subject.to_html.should == "<li>Root<ul><li>Child</li></ul></li>"
+        end
+
+        it "wraps children in the passed list element" do
+          subject.list_element_pair = [:nav, :div]
+          subject.to_html.should == "<div>Root<nav><div>Child</div></nav></div>"
```
Comment (knewter) [inline delivery/rails/spec/presenters/html/menu_entry_presenter_spec.rb:44]: :heart: I know this was already in here but hadn't said how much I liked this being a thing :)

**Comment by knewter** [comment]
I see nothing but awesomeness here.  :+1:

File: delivery/rails/spec/presenters/html/menu_entry_presenter_spec.rb (hunk 2 above)
Comment (robyurkowski) [inline delivery/rails/spec/presenters/html/menu_entry_presenter_spec.rb:44]: It does make life massively easier. Once we start looking at potential use cases, it'd shock me if we didn't make the class that you use for rendering menu child objects configurable, too, so you could just subclass `MenuEntry` and have all your children rendered a special way, just by setting the option on `Menu`.

In fact, I think when I make `Menu` work a bit better (atm it doesn't wrap its child items in presenters as it should), this'll happen. Then it'll be +5 awesomeness.

**Comment by robyurkowski** [comment]
Hearing no objections...

**Pull request merged**